src/repository-classes.ts

```diff
@@ -0,0 +1,110 @@
+import * as core from '@actions/core';
+import * as semver from 'semver';
+import { ACCESS_TOKEN } from './config';
+import { Octokit } from '@octokit/rest';
+import { SlackMessage } from './message';
+import fetch from 'node-fetch';
+
+interface IGithubFileContent {
+    content: string;
+}
+
+interface INecessaryDataFromManifest {
+    version: string;
+}
+
+const octokit = new Octokit({ 
+    auth: `${ACCESS_TOKEN}`,
+    request: {
+        fetch: fetch,
+    },
+});
+
+abstract class BaseRepository {
+    constructor (public owner: string, public repo: string) {
+        this.owner = owner;
+        this.repo = repo;
+    }
+}
+
+export class InternalRepository extends BaseRepository {
+    constructor (owner: string = 'dusan-trickovic', repo: string = 'automation-tests') {
+        super(owner, repo);
+    }
+}
+
+export class ManifestRepository extends BaseRepository {
+    constructor (owner: string, repo: string, public path: string = 'versions-manifest.json') {
+        super(owner, repo);
+    }
+
+    async getVersionsManifestFromRepo(referenceVersion: string): Promise<INecessaryDataFromManifest[]> {
+        try {   
+            const response = await octokit.repos.getContent({
+            owner: this.owner,
+            repo:  this.repo,
+            path:  this.path,
+            });
+    
+            const githubFileContent = response.data as IGithubFileContent;
+            const content = Buffer.from(githubFileContent.content, 'base64').toString();
+            const jsonData = JSON.parse(content);
+            const reversedJsonData = jsonData.reverse();
+            const latestFromManifest: INecessaryDataFromManifest[] = reversedJsonData.filter((item: INecessaryDataFromManifest) => {
+                return semver.gte(item.version, referenceVersion);
+            });
+    
+            return latestFromManifest as INecessaryDataFromManifest[];
+        } catch (error) {
+            core.setFailed((error as Error).message);
+            return [];
+        }
```

Comment:
nitpick: I think there is no need to return something after `setFailed`
```suggestion
        } catch (error) {
            core.setFailed((error as Error).message);
        }
```

Reply:
@MaksimZhukov I mostly added it because of a typed data to be returned by the function, but I suppose I can just add a `throw new Error()` to handle that problem?

Reply:
Sounds good to me

Reply:
Done